core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -103,7 +105,7 @@ private[spark] class EventLoggingListener(
 
   // Events that do not trigger a flush
   override def onStageSubmitted(event: SparkListenerStageSubmitted): Unit = {
-    logEvent(event)
+    logEvent(SparkListenerStageSubmitted(event.stageInfo, redactProperties(event.properties)))
```

Comment:
Use copy instead - here and elsewhere, to be forward compatible.
For example: `event.copy(properties =  redactProperties(event.properties))`